pkg/controller/flink/client/error_handler.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// appError codes
+const (
+	globalFailure      = "FAILED"
+	jsonUnmarshalError = "JSONUNMARSHALERROR"
+	defaultRetries     = 5
```

Comment:
I think this, combined with the 100ms backoff, is too low. I think this gives a retry schedule like

```
0	100  	100
1	200  	300
2	400 	700
3	800 	1500
4	1600	3100
```
(count, backoff ms, cumulative ms)

So max in total we only wait 3 seconds. I recommend something like 20 retries which (combined with a bound on the backoff) could give a failing external resource time to recover.

Reply:
Sounds good. I'll increase the number of retries here.